include/gpiod.h

```diff
@@ -1010,6 +1032,52 @@ const char *
 gpiod_chip_iter_failed_chip(struct gpiod_chip_iter *iter) GPIOD_API;
 
 /**
+ * @brief Iterator structure for GPIO chip bulk objects.
+ */
+struct gpiod_chip_bulk_iter
+{
+	struct gpiod_chip_bulk *bulk;
+	unsigned int index;
+};
+
+/**
+ * @brief Create a new gpiochip bulk iterator.
+ * @return Pointer to a new chip iterator object or NULL if an error occurred.
+ *
+ * This iterator is used to walk over the opened chips of a bulk object without
+ * closing them.
+ */
+struct gpiod_chip_bulk_iter *
+gpiod_chip_bulk_iter_new(struct gpiod_chip_bulk *bulk) GPIOD_API;
+
+/**
+ * @brief Get the next gpiochip handle from the bulk object.
+ * @param iter The gpiochip bulk iterator object.
+ * @return Pointer to an open gpiochip handle or NULL if the next chip can't
+ *         be accessed.
+ *
+ * Internally this routine walks over the previously opened chips of the bulk
+ * object. The returned chips are kept open. If an error occurs or no more
+ * chips are present, the function frees the iterator and returns NULL.
+ */
+struct gpiod_chip *
+gpiod_chip_bulk_iter_next(struct gpiod_chip_bulk_iter *iter) GPIOD_API;
```

Comment:
If `zalloc` in `gpiod_chip_bulk_iter_new` returns NULL, it needs to be propagated through `gpiod_chip_bulk_iter_next` to cancel the foreach loop and avoid a NULL ptr dereference.